scripts/tutorials/tutorials.lisp

```diff
@@ -8,20 +8,26 @@
 
 ;; NOTE: the list should not exceed 64 tutorials.
 
-'(;; ("template" "tutorials/template.lisp" 0)
-  ("Introduction" "tutorials/overview.lisp" 0)
-  ("Game Speed" "tutorials/game_speed.lisp" 1)
-  ("Weapon Groups" "tutorials/weapon_groups.lisp" 2)
-  ("Firing Patterns" "tutorials/firing_patterns.lisp" 3)
-  ("Target Queueing" "tutorials/target_queueing.lisp" 4)
-  ("Salvage" "tutorials/salvage.lisp" 5)
-  ("Moving Blocks" "tutorials/moving_blocks.lisp" 6)
-  ("Construction Tricks" "tutorials/construction_tricks.lisp" 7)
-  ("Advanced Structures" "tutorials/advanced_structures.lisp" 8)
-  ("Your Crew"  "tutorials/characters.lisp" 9)
-  ("Replicators" "tutorials/replicator.lisp" 10)
-  ("Transporters" "tutorials/transporter.lisp" 11)
-  ("Portals" "tutorials/portals.lisp" 15)
-  ("Adding Terrain" "tutorials/terrain.lisp" 12)
-  ("Power Balance" "tutorials/power_balance.lisp" 13)
-  ("Diverting Power" "tutorials/power_balance_2.lisp" 14))
+(setq tutorials-list
+  '(
+    ;; Tutorials list
+    ("Introduction" "tutorials/overview.lisp" 0)
+    ("Game Speed" "tutorials/game_speed.lisp" 1)
+    ("Weapon Groups" "tutorials/weapon_groups.lisp" 2)
+    ("Firing Patterns" "tutorials/firing_patterns.lisp" 3)
+    ("Target Queueing" "tutorials/target_queueing.lisp" 4)
+    ("Salvage" "tutorials/salvage.lisp" 5)
+    ("Moving Blocks" "tutorials/moving_blocks.lisp" 6)
+    ("Construction Tricks" "tutorials/construction_tricks.lisp" 7)
+    ("Advanced Structures" "tutorials/advanced_structures.lisp" 8)
+    ("Your Crew" "tutorials/characters.lisp" 9)
+    ("Replicators" "tutorials/replicator.lisp" 10)
+    ("Transporters" "tutorials/transporter.lisp" 11)
+    ("Portals" "tutorials/portals.lisp" 15)
+    ("Adding Terrain" "tutorials/terrain.lisp" 12)
+    ("Power Balance" "tutorials/power_balance.lisp" 13)
+    ("Diverting Power" "tutorials/power_balance_2.lisp" 14)
+  ))
+
+(if (is-developer-mode)
+    (setq tutorials-list (append tutorials-list '(("Template" "tutorials/template.lisp" 15)))))
```

Comment:
This is a neat idea! But I think that code actually won't work outside of developer mode.
The script is supposed to return the tutorials list. In developer mode, it returns the result of setq, which is the tutorials list, but when not in developer mode, the if expression evaluates the empty false condition and returns nil.

We can do the same thing without any intermediate variables:
``` lisp

(append '(("Introduction" "tutorials/overview.lisp" 0)
          ("Game Speed" "tutorials/game_speed.lisp" 1)
          ("Weapon Groups" "tutorials/weapon_groups.lisp" 2)
          ("Firing Patterns" "tutorials/firing_patterns.lisp" 3)
          ("Target Queueing" "tutorials/target_queueing.lisp" 4)
          ("Salvage" "tutorials/salvage.lisp" 5)
          ("Moving Blocks" "tutorials/moving_blocks.lisp" 6)
          ("Construction Tricks" "tutorials/construction_tricks.lisp" 7)
          ("Advanced Structures" "tutorials/advanced_structures.lisp" 8)
          ("Your Crew"  "tutorials/characters.lisp" 9)
          ("Replicators" "tutorials/replicator.lisp" 10)
          ("Transporters" "tutorials/transporter.lisp" 11)
          ("Portals" "tutorials/portals.lisp" 15)
          ("Adding Terrain" "tutorials/terrain.lisp" 12)
          ("Power Balance" "tutorials/power_balance.lisp" 13)
          ("Diverting Power" "tutorials/power_balance_2.lisp" 14))
        (if (is-developer-mode) ; In developer mode, append t
            '(("template" "tutorials/template.lisp" 16))
            '()))
```

It's better to avoid creating the tutorials-list global variable, because it the tutorials list will sit in memory. If you want to assign it to a variable, it's better to use a let expression, so that the variable will be cleaned up after this script finishes running.
Furthermore, let's reserve a tutorial number for this template file. 15 is in use for the portals level, how about 16? And create an empty test/16.lisp file for regression.

Reply:
I'm surprised that the interpreter allows a setq expression for an undeclared global variable tutorials-list, but regression catches the problem, at least. I'll look into that at some point.

Reply:
> 15 is in use for the portals level

genuinely forgot about the portals tutorial